src/player.cpp

```diff
@@ -1898,21 +1898,26 @@ player::wear( item_location item_wear, bool interactive )
     return result;
 }
 
+int player::adjust_lift_str( int str ) const
+{
+    if( has_trait( trait_id( "STRONGBACK" ) ) ) {
+        str *= 1.35;
+    } else if( has_trait( trait_id( "BADBACK" ) ) ) {
+        str /= 1.35;
+    }
+    return str;
+}
+
 template <typename T>
 bool player::can_lift( const T &obj ) const
 {
     // avoid comparing by weight as different objects use differing scales (grams vs kilograms etc)
-    int str = get_str();
+    int str = adjust_lift_str( get_str() );
     if( mounted_creature ) {
         const auto mons = mounted_creature.get();
         str = mons->mech_str_addition() == 0 ? str : mons->mech_str_addition();
     }
     const int npc_str = get_lift_assist();
-    if( has_trait( trait_id( "STRONGBACK" ) ) ) {
```

Comment:
Thank you for updating. I'm sorry if I didn't explain clearly last time. `str` can be altered in `if(mounted_creature)` block before adjusting for `STRONGBACK`, so you need to follow the way the original code adjusted strength, that is, adjust strength after `if( mounted_creature )`.